staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -292,6 +292,9 @@ func (c *threadSafeMap) deleteFromIndices(obj interface{}, key string) {
 			set := index[indexValue]
 			if set != nil {
 				set.Delete(key)
+				if set.Len() == 0 {
+					delete(index, indexValue)
```

Comment:
not sure if it's a valid fix. otoh, this could increase the overhead of memory [allocation|gc]. practically, the index aligns w/ namespace which doesn't seem to be sth we're frequently creating/deleting?